hw/platform/snowy_family/snowy_qemu.c

```diff
@@ -0,0 +1,60 @@
+#include "snowy_qemu.h"
+#include "stm32_usart.h"
+#include "stm32_power.h"
+#include "semphr.h"
+
+static const stm32_usart_config_t _usart2_config = {
+    .usart                = USART2,
+    .flow_control_enabled = FLOW_CONTROL_DISABLED,
+    .usart_periph_bus     = STM32_POWER_APB1,
+    .gpio_pin_tx_num      = 2,
+    .gpio_pin_rx_num      = 3, // this is even more ignored by QEMU than tx
+    .gpio_pin_rts_num     = 0,
+    .gpio_pin_cts_num     = 0,
+    .gpio_ptr             = GPIOA,
+    .gpio_clock           = RCC_AHB1Periph_GPIOA,
+    .usart_clock          = RCC_APB1Periph_USART2,
+    .af                   = GPIO_AF_USART2,
+};
+
+static stm32_usart_t _usart2 = {
+    &_usart2_config,
+    NULL, /* no dma */
+    230400
+};
+
+static StaticSemaphore_t _usart2_mutex_mem;
```

Comment:
I know this is very WIP, but as a general rule any RTOS specific code such as a mutex/semaphore thread etc live in /rcore

This allows for a little clearer separation and when it comes to getting tintin ready it will make this a bit easier.

I'm sure you know already, but just a gentle tap

Reply:
I'll clarify in that locking usually happens in rcore. in this case rcore/qemu.c looks likely